### PR TITLE
Separate go mod for e2e (E2E #4)

### PR DESCRIPTION
**Pull request by chatton**
## Description

This PR removes `ibctest` as a dependency of the `ibc-go` module and creates a seperate go.mod file for our e2e tests.

The E2E specific Makefile commands have been extracted and moved into a separate Makefile under the `e2e` directory.

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: e2e/Makefile [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+TEST_CONTAINERS=$(shell docker ps --filter "label=ibc-test" -a -q)
+
+cleanup-ibc-test-containers:
+	for id in $(TEST_CONTAINERS) ; do \
+		$(DOCKER) stop $$id ; \
+		$(DOCKER) rm $$id ; \
+	done
+
+e2e-test: cleanup-ibc-test-containers
+	@go test -v ./ --run $(suite) -testify.m ^$(test)$$
```
Comment (damiannolan) [inline e2e/Makefile:11]: I'm not exactly a Makefile guru, but should these `.PHONY` targets as they're not creating an output file?

Comment (chatton) [reply e2e/Makefile:11]: you are absolutely correct, thanks for spotting that!

File: e2e/go.mod [hunk 2]
```diff
@@ -0,0 +1,142 @@
+module e2e
```
Comment (colin-axner) [inline e2e/go.mod:1]: Is this common practice? I would think the import should be "github.com/cosmos/ibc-go/e2e"? What would the differences be between using the different module paths?

Comment (chatton) [reply e2e/go.mod:1]: I was initially thinking of to do something like this, but opted to remove the `ibc-go` prefix to make it clear it is a separate entity. If you think it's better to have the fully qualified name we can always rename it.

**Pull request merged**